script.js

```diff
@@ -1,4 +1,275 @@
+  }
+};
+
+//Create a function that resets the game
+var gameReset = function () {
+  hasPlayer1Rolled = 0;
+  hasPlayer2Rolled = 0;
+  player1DiceRolls = [];
+  player2DiceRolls = [];
+  player1ConcatenatedNumber = 0;
+  player2ConcatenatedNumber = 0;
+  timeToReset = 0;
+  return `The game has been reset! You may play again by pressing the "Submit" button.`;
+};
+
+//Create a function that receives the user input and returns the correct output statements
+var beatThatGame = function (userInput) {
+  //Make the input entirely lowercase
+  userInput = userInput.toLowerCase();
+
+  if (
+    !(userInput == "dice 1" || userInput == "dice 2") &&
+    ((hasPlayer1Rolled == 1 && player1ConcatenatedNumber == 0) ||
+      (hasPlayer1Rolled == 1 &&
+        hasPlayer2Rolled &&
+        player2ConcatenatedNumber == 0))
+  ) {
+    return;
+  }
+  if (userInput == "" && !(hasPlayer1Rolled == 1 && hasPlayer2Rolled == 1)) {
+    //Check if there is no input, which means the players are rolling the dice. Call the dice rolling functions
+    rollTheDice();
+    return statementForDiceRolled();
+  }
+
+  //Check if input is "dice 1" or "dice 2", which means the players are choosing which dice is the first number and which is the second. Call the concatenating functions
+  if (userInput == "dice 1" || userInput == "dice 2") {
+    concatenator(userInput);
+    return statementForConcatenation(userInput);
+  }
+
+  //Check if input is "compare", which means its time to compare the two numbers and see who won
+  if (
+    userInput == "compare" &&
+    hasPlayer1Rolled == 1 &&
+    hasPlayer2Rolled == 1
+  ) {
+    timeToReset = 1;
+    return comparingWinnerStatement();
+  }
+
+  //Check if input is "reset", which means that it is time to reset the game
+  if (userInput == "reset" && timeToReset == 1) {
+    return gameReset();
+  }
+  console.log("Player 1: ", player1DiceRolls, "Player 2: ", player2DiceRolls);
+};
+
+//Create a function that takes the user input and outputs the correct instructions if the user types invalid input
+var invalidInputChecker = function () {
+  //Check if both players have not rolled the dice yet
+  if (hasPlayer1Rolled == 0 && hasPlayer2Rolled == 0) {
+    return `Please press the "submit" button without typing anything in the input box to start the game for player 1`;
+  }
+
+  if (
+    hasPlayer1Rolled == 1 &&
+    hasPlayer2Rolled == 0 &&
+    player1ConcatenatedNumber == 0
+  ) {
+    return `Player 1, please select which dice you would like to be in front. <br><br>You rolled a ${player1DiceRolls[0]} for Dice 1 and a ${player1DiceRolls[1]} for Dice 2. <br><br> Input "dice 1" for Dice 1 or "dice 2" for Dice 2`;
+  }
+
+  if (hasPlayer1Rolled == 1 && hasPlayer2Rolled == 0) {
+    return `Please press the "submit" button without typing anything in the input box to start the game for player 2`;
+  }
```

Comment:
same comment as above, do we need this odd constraint of the input box being empty?